examples/48-drawindirect/drawindirect.cpp

```diff
@@ -0,0 +1,381 @@
+				if (ii % 2) {
+					// Tetrahedron
+					objs[ii].m_vertexOffset = 8;
+					objs[ii].m_vertexCount = 4; // m_vertexCount is unused in compute shader, its only here for completeness
+					objs[ii].m_indexOffset = 36;
+					objs[ii].m_indexCount = 12;
+					}
+				else {
+					// Cube
+					objs[ii].m_vertexOffset = 0;
+					objs[ii].m_vertexCount = 8;
+					objs[ii].m_indexOffset = 0;
+					objs[ii].m_indexCount = 36;
+					}
+				}
+			
+			// This is a list of objects to be rendered via the indirect program
+			m_object_list_buffer = bgfx::createVertexBuffer(bgfx::copy(objs, sizeof(objs) ), ObjectInstance::ms_layout, BGFX_BUFFER_COMPUTE_READ);
+			
+			// This is the instance buffer used for rendering. 
+			// You could instead use a dynamic instance buffer when rendering (use bgfx::allocInstanceDataBuffer in draw loop)
+			// Fill it with a blank array for now, we will populate it with compute
+			RenderInstance temp[m_nDrawElements];
+			memset(&temp[0], 0, sizeof(temp));
+			m_instance_buffer = bgfx::createVertexBuffer(bgfx::copy(temp, sizeof(temp) ), RenderInstance::ms_layout, BGFX_BUFFER_COMPUTE_WRITE);
+			}
+			
+		m_timeOffset = bx::getHPCounter();
+
+		imguiCreate();
+	}
+
+	int shutdown() override
+	{
+		imguiDestroy();
+
+		// Cleanup.
+		bgfx::destroy(m_ibh);
+		bgfx::destroy(m_vbh);
+		bgfx::destroy(m_program);
+
+		if (bgfx::isValid(m_indirect_program))
+			bgfx::destroy(m_indirect_program);
+		if (bgfx::isValid(m_indirect_buffer_handle))
+			bgfx::destroy(m_indirect_buffer_handle);
+		if (bgfx::isValid(m_object_list_buffer))
+			bgfx::destroy(m_object_list_buffer);
+		if (bgfx::isValid(m_instance_buffer))
+			bgfx::destroy(m_instance_buffer);
+		bgfx::destroy(u_drawParams);
+		
+		// Shutdown bgfx.
+		bgfx::shutdown();
+
+		return 0;
+	}
+
+	bool update() override
+	{
+		if (!entry::processEvents(m_width, m_height, m_debug, m_reset, &m_mouseState) )
+		{
+			imguiBeginFrame(m_mouseState.m_mx
+				,  m_mouseState.m_my
+				, (m_mouseState.m_buttons[entry::MouseButton::Left  ] ? IMGUI_MBUT_LEFT   : 0)
+				| (m_mouseState.m_buttons[entry::MouseButton::Right ] ? IMGUI_MBUT_RIGHT  : 0)
+				| (m_mouseState.m_buttons[entry::MouseButton::Middle] ? IMGUI_MBUT_MIDDLE : 0)
+				,  m_mouseState.m_mz
+				, uint16_t(m_width)
+				, uint16_t(m_height)
+				);
+
+			showExampleDialog(this);
+
+			imguiEndFrame();
+			
+			// Get renderer capabilities info.
+			const bool computeSupported = !!(BGFX_CAPS_DRAW_INDIRECT & bgfx::getCaps()->supported);
+			const bool instancingSupported = !!(BGFX_CAPS_INSTANCING & bgfx::getCaps()->supported);
+
+			// Set view 0 default viewport.
+			bgfx::setViewRect(0, 0, 0, uint16_t(m_width), uint16_t(m_height) );
+
+			// This dummy draw call is here to make sure that view 0 is cleared
+			// if no other draw calls are submitted to view 0.
+			bgfx::touch(0);
+
+			float time = (float)( (bx::getHPCounter() - m_timeOffset)/double(bx::getHPFrequency() ) );
+
+			if (!(computeSupported && instancingSupported))
```

Comment:
Move this block as else below, you're already testing for compute/instancing support.